controllers/restore_controller.go

```diff
@@ -268,20 +266,36 @@ func (r *RestoreReconciler) initVeleroRestore(ctx context.Context, restore *v1be
 
 // attachManagedClusters is the entry points for all the managed clusters to be re-attached
 // currently it loops across all the managed cluster namespaces
-func (r *RestoreReconciler) attachManagedClusters(ctx context.Context, restore *v1beta1.Restore) (bool, error) {
+func (r *RestoreReconciler) attachManagedClusters(ctx context.Context, restore *v1beta1.Restore, restoreName string) (bool, error) {
 	allAttached := false
-	namespaceList := v1.NamespaceList{}
-	labelSelector, _ := labels.Parse(OCMManagedClusterNamespaceLabelKey)
-	if err := r.Client.List(ctx, &namespaceList, &client.ListOptions{LabelSelector: labelSelector}); err != nil {
-		return allAttached, fmt.Errorf("unable to list managed cluster namespaces: %v", err)
+
+	// find all hive.openshift.io kubeconfig secrets found with the current restore file
+	// this gives the list of managed clusters namespaces to be processed by this restore op
+	kubeAdminSecrets := getAllKubeAdminSecrets(ctx, r.Client, restoreName)
+
+	if kubeAdminSecrets == nil {
+		return allAttached, fmt.Errorf("unable to retrieve managed cluster admin secrets")
 	}
+
 	var errors []error
 	allAttached = true
-	for i := range namespaceList.Items {
-		if namespaceList.Items[i].Name == "local-cluster" {
+
+	proccessedClusters := []string{}
+	for i := range kubeAdminSecrets.Items {
+		// ignore local cluster
+		if kubeAdminSecrets.Items[i].Namespace == "local-cluster" {
 			continue
 		}
-		attached, err := r.attachManagedCluster(ctx, restore, namespaceList.Items[i].Name)
+
+		// ignore already processed namespaces, assuming we have more then one kube admin secret in the same ns
+		_, ok := find(proccessedClusters, kubeAdminSecrets.Items[i].Namespace)
+		if ok {
+			// this namespace was already processed, ignore it
+			continue
+		}
+		proccessedClusters = append(proccessedClusters, kubeAdminSecrets.Items[i].Namespace)
```

Comment:
I would implement differently, two things here:
1. We can set a namespaces has been processed as soon we succesfully run `attachManagedCluster` so the mechanism to _tag_ a namespace `processed` should go AFTER trying to `attachManagedCluster`.
2. The mechanism to _tag_ a cluster has `attached` to the new HUB should be persisted in the `Status` of the restore. I would say throuhg a `Condition`.

Reply:
Dario, let's put this on the back burner until we get the ManagedCluster CR into the backup
We'll decide then how to set the labels and where; in that case we can probably make use of the ManagedCluster resource